blogposts/2021/eliminating-secrets-from-codebase.md

```diff
@@ -0,0 +1,134 @@
+---
+title: How we used universal code search to eliminate secrets from our codebase
+description: "We used to store secrets in our source code. Here’s how we used code search to help us find, remove, and rotate all secrets in our codebase."
+author: André Eleuterio
+authorUrl: https://about.sourcegraph.com/handbook/company/team#andr%c3%a9-eleuterio-he-him
+publishDate: 2021-09-27T10:00-07:00
+tags: [blog]
+slug: eliminate-secrets-from-codebase-with-universal-code-search
+heroImage: https://sourcegraphstatic.com/blog/securing-sourcegraph-eliminating-secrets.webm
+socialImage: https://sourcegraphstatic.com/blog/securing-sourcegraph-eliminating-secrets.mp4
+published: true
+---
+
+<video loop autoplay muted playsinline>
+  <source src="https://sourcegraphstatic.com/blog/securing-sourcegraph-eliminating-secrets.webm" type="video/webm" />
+  <source src="https://sourcegraphstatic.com/blog/securing-sourcegraph-eliminating-secrets.mp4" type="video/mp4" />
+</video>
```

Comment:
@felixfbecker I'm hoping you can help me here! I tried to follow your directions in https://about.sourcegraph.com/handbook/marketing/adding_screenshots_screen_recording#embedding-gif-like-videos but either I've done something incorrectly or it doesn't quite work for this use case. We have a gif/gif-like video to use for the blog thumbnail, social, and hero image, but as it stands it's not rendering on the card on [/blog](https://deploy-preview-4791--sourcegraph.netlify.app/blog) and [in the post itself](https://deploy-preview-4791--sourcegraph.netlify.app/blog/eliminate-secrets-from-codebase-with-universal-code-search/) the video is too big and cut off on one side. You can see a previous example of using a .gif [here](https://about.sourcegraph.com/blog/role-of-observability-tools-high-performing-development-teams/). Any help would be much appreciated!

Reply:
Happy to help! There are a couple different problems here, let me address them separately:

For the hero image in the blog post, I can see that the blog CSS has explicit CSS that gives _images_ a `max-width` of 75% (and also margin, a box-shadow, etc), while videos have no CSS styles at all and therefore overflow the container. This needs to be fixed by adding CSS styles for `video`s, but it's a question to the marketing/design team what you want the width to be? E.g. 100%, 75% just like images, a specific max-width of pixels/rems (e.g. paragraphs are limited to 40rems = 640px width), etc. Happy to quickly add that change.

For the social image and hero image, I want to ask whether you're sure that you want to use an animation for this. It could potentially be annoying for the hero images to have animations dance around everywhere (of course, it can also be eye-catching).

But the social image actually won't support animations for some social media platforms. E.g. [Twitter supports the GIF format, but it will only display the first frame](https://developer.twitter.com/en/docs/twitter-for-websites/cards/overview/summary-card-with-large-image), meaning it's not animated and it's probably not what you want for this animation because the robots won't be visible, only the chest. The same can be true for other social media platforms too, I'm not sure, but Twitter seems like the most important one for the dev community to me.

I'd recommend exporting the last frame (or whichever you want to use) of the animation as a still image (PNG) and then using that as the social and hero image.

Reply:
I pushed a quick change to this branch to simply apply the same styles we have for images (including "real" GIFs) to GIF-like videos too.

Reply:
@felixfbecker Thank you so much! I expect some of these things will be addressed with the overall blog redesign, but thank you for pushing the fix to make the video work in the post itself in the meantime.

I had forgotten that gifs don't work for the social image so I can swap a still in for that! 

For the hero, I'm assuming the blog posts with gifs as heroes work right now (as in [this post](https://about.sourcegraph.com/blog/role-of-observability-tools-high-performing-development-teams/)) because it's an actual .gif file rather than a video?

Reply:
Yeah you can technically use an animated GIF for the hero, but I would still question whether that is a good thing for the blog landing page (imagine all hero imagines dancing around).

If that _is_ something the marketing team wants to have, then we could also support "hero animations" with some code changes (we control how we render these images, and can conditionally render a `<video loop autoplay muted>`.).

Reply:
@felixfbecker Ok thanks for confirming! I agree we would want to be selective about it. I will add this to the list of considerations for the redesign 🎉